src/credentials/loginManager.ts

```diff
@@ -91,3 +98,71 @@ export class LoginManager {
         await this.awsContext.setCredentials(undefined)
     }
 }
+
+/**
+ * Connects last-used AWS credentials, if not already attempted in the current session.
+ *
+ * @returns true if login succeeded or credentials exist; false if login failed
+ * or was already tried this session.
+ */
+export async function tryAutoLogin(awsContext: AwsContext): Promise<boolean> {
+    let creds = await awsContext.getCredentials() // Current credentials?
+    if (creds) {
+        return true
+    }
+    if (didTryAutoConnect) {
+        return false
+    }
+    didTryAutoConnect = true
+    const toolkitSettings = new DefaultSettingsConfiguration(extensionSettingsPrefix)
+    const loginManager = new LoginManager(awsContext, new CredentialsStore())
+    await loginWithMostRecentCredentials(toolkitSettings, loginManager)
+    creds = await awsContext.getCredentials() // Connected credentials?
+    return !!creds
```

Comment:
Would this be a little faster if you just return a boolean from `loginWithMostRecentCredentials` denoting success/failure?

Reply:
I intentionally didn't do that because the conditions where loginWithMostRecentCredentials returns false are not clear, not documented, and could change in the future. So this is more reliable.